---
title: "run Dsuite on Phylloscopus"
format: html
toc: true
toc-title: Document Contents
number-sections: true
embed-resources: true
---

### Prepare input files

For Dsuite input, we will need to prepare 3 files:

1. A vcf file containing SNP data
  - We can just use our filtered, unlinked SNP dataset as input &#9745;
2. A tree file containing a pre-specified topology
  - The following string shows the major genome-wide topology in newick format: '((((amoenus),(pallescens)),(makira,(guad,(malaita,isabel)))),Outgroup);'. This can be saved to the file nwk.tre &#9745;
3. A pops file assigning samples to populations
  - This just looks like:
  
    P_amoenus_279.sorted	amoenus
    P_amoenus_36181.sorted	amoenus
    P_amoenus_36184.sorted	amoenus
    P_amoenus_36208.sorted	amoenus
    P_amoenus_DOT251.sorted	amoenus
    P_becki_218146.sorted	isabel
    P_becki_227294.sorted	malaita
    P_becki_32802.sorted	guad
    P_becki_32814.sorted	guad
    P_giulianettii_16583.sorted	Outgroup
    P_giulianettii_27874.sorted	Outgroup
    P_makirensis_35004.sorted	makira
    P_makirensis_35012.sorted	makira
    P_pallescens_220.sorted	pallescens
    P_pallescens_266.sorted	pallescens
    P_pallescens_36191.sorted	pallescens
    P_pallescens_36192.sorted	pallescens
    P_pallescens_36200.sorted	pallescens
    
    This can be saved to a file called 'pops.txt' &#9745;

### Now, run Dsuite
```{bash, eval=FALSE}
#this command uses the program Dsuite (v0.4) to calculate pairwise f-statistics across a tree.
/home/d669d153/work/Dsuite/Build/Dsuite Dtrios -c -n geneflow.top -t nwk.tre phyllo.autosomal.filtered.1kbthinned.SNPs.vcf.gz pops.txt
```

### Calculate fbranch and default visualizations
```{bash, eval=FALSE}
#make a 'plot order' text file. Mine just looks like this:
isabel
malaita
guadalcanal
makira
pallescens
amoenus
Outgroup

##copy in ruby scripts. I already have them downloaded in another directory. 
cp /home/d669d153/work/todi.subset.2022/dsuite/plot_d.rb .
cp /home/d669d153/work/todi.subset.2022/dsuite/plot_f4ratio.rb .

#If you don't, you can download them with:
wget https://raw.githubusercontent.com/millanek/tutorials/master/analysis_of_introgression_with_snp_data/src/plot_d.rb
wget https://raw.githubusercontent.com/millanek/tutorials/master/analysis_of_introgression_with_snp_data/src/plot_f4ratio.rb

##plot heatmap of D
#the text file is an input file you generated by running Dtrios, that will always end with '_BBAA.txt'
ruby plot_d.rb pops_geneflow.top_BBAA.txt plotorder.txt 0.7 geneflow_BBAA_D.svg

##plot heatmap of f4
ruby plot_f4ratio.rb pops_geneflow.top_BBAA.txt plotorder.txt 0.2 geneflow_BBAA_f4ratio.svg

##run fbranch on the guidetree, with the Dtrios output ending in 'tree.txt' (guidetree must be rooted on #outgroup)
/home/d669d153/work/Dsuite/Build/Dsuite Fbranch nwk.tre pops_geneflow.top_tree.txt > geneflow_Fbranch.txt

#activate python
module load python
#plot the fbranch heatmap with guidetree above
/home/d669d153/work/Dsuite/utils/dtools.py geneflow_Fbranch.txt nwk.tre

#rerun Fbranch to generate output with corresponding Z scores
/home/d669d153/work/Dsuite/Build/Dsuite Fbranch nwk.tre pops_geneflow.top_tree.txt -Z T > geneflow_Fbranch.txt
```


```{r}
#show the final product
#fbranch statistics with Z scores labeled
knitr::include_graphics("~/Desktop/one.png")
```

Now we can repeat the exact same procedure, except this time specifying the alternate tree, supported by just 10% of gene trees. To do this, we will just make a new directory, copy in the vcf and the popmap, and specify a new tree named 'nwk.tre', that looks like this: '((amoenus,(pallescens,(((isabel,malaita),guad),makira))),Outgroup);'

### Re-run with an alternate topology
```{bash, eval=FALSE}
#this command uses the program Dsuite (v0.4) to calculate pairwise f-statistics across a tree.
/home/d669d153/work/Dsuite/Build/Dsuite Dtrios -c -n bifurcating.top -t nwk.tre phyllo.autosomal.filtered.1kbthinned.SNPs.vcf.gz pops.txt
```

### re-do fbranch and default visualizations
```{bash, eval=FALSE}
#bring in plotorder.txt
cp /home/d669d153/work/Wphyllowgs/dsuite/autosomal/geneflow.top/plotorder.txt .

##copy in ruby scripts. I already have them downloaded in another directory. 
cp /home/d669d153/work/todi.subset.2022/dsuite/plot_d.rb .
cp /home/d669d153/work/todi.subset.2022/dsuite/plot_f4ratio.rb .

##plot heatmap of D
#the text file is an input file you generated by running Dtrios, that will always end with '_BBAA.txt'
ruby plot_d.rb pops_bifurcating.top_BBAA.txt plotorder.txt 0.7 bifurcating_BBAA_D.svg

##plot heatmap of f4
ruby plot_f4ratio.rb pops_bifurcating.top_BBAA.txt plotorder.txt 0.2 bifurcating_BBAA_f4ratio.svg

##run fbranch on the guidetree, with the Dtrios output ending in 'tree.txt' (guidetree must be rooted on #outgroup)
/home/d669d153/work/Dsuite/Build/Dsuite Fbranch nwk.tre pops_bifurcating.top_tree.txt > bifurcating_Fbranch.txt

#activate python
module load python
#plot the fbranch heatmap with guidetree above
/home/d669d153/work/Dsuite/utils/dtools.py bifurcating_Fbranch.txt nwk.tre

##rerun fbranch to make an output file with corresponding z scores
/home/d669d153/work/Dsuite/Build/Dsuite Fbranch nwk.tre pops_bifurcating.top_tree.txt -Z T > bifurcating_Fbranch.txt

```


```{r}
#show the final product
#fbranch statistics with Z scores labeled
knitr::include_graphics("~/Desktop/two.png")
```


### Add in Z chromosome investigation
```{bash}

```

